src/KlaviyoAPI.php

```diff
@@ -183,7 +183,8 @@ private function handleResponse( $response, $statusCode, $isPublic )
                 $this->returnRateLimit( $this->decodeJsonResponse( $response ) )
             );
         } else if ( $statusCode != 200 ) {
-            throw new KlaviyoException( sprintf( self::ERROR_NON_200_STATUS, $statusCode ) );
+            $responseDetail = $statusCode . ' (' . $this->decodeJsonResponse( $response )['detail'] . ')';
```

Comment:
Would be keen on what others think here, but I think this should be a new exception

Inheritance structure:
KlaviyoException -> KlaviyoApiException -> KlaviyoAuthenticationException / KlaviyoResourceNotFoundException / KlaviyoRateLimitException

`KlaviyoApiException` Would then look something like: `KlaviyoApiException( status code, responseDetails, response )`

Reply:
I'm cool with that.